digicamtoy/tracegenerator.py

```diff
@@ -36,13 +36,56 @@ def _generalized_poisson_crosstalk(crosstalk, n_photon):
 
 class NTraceGenerator:
 
-    def __init__(self,
-                 time_start=0, time_end=200, time_sampling=4, n_pixels=1296,
-                 nsb_rate=0.6, crosstalk=0.08, gain_nsb=True, n_photon=0,
-                 poisson=True, sigma_e=0.8, sigma_1=0.8, gain=5.8,
-                 baseline=200, time_signal=20, jitter=0,
-                 pulse_shape_file='/utils/pulse_SST-1M_pixel_0.dat',
-                 sub_binning=0, **kwargs):
+    def __init__(
+        self,
+        time_start=0,
+        time_end=200,
+        time_sampling=4,
+        n_pixels=1296,
+        nsb_rate=0.6,
+        crosstalk=0.08,
+        gain_nsb=True,
+        n_photon=0,
+        poisson=True,
+        sigma_e=0.8,
+        sigma_1=0.8,
+        gain=5.8,
+        baseline=200,
+        time_signal=20,
+        jitter=0,
+        pulse_shape_file='/utils/pulse_SST-1M_pixel_0.dat',
+        sub_binning=0,
+        **kwargs
+    ):
+        '''
+        time_start: scalar
+            start time of sampled waveforms (in ns?).
+
+        time_end: scalar
+            end time of sampled waveforms in the same unit as time_start.
+
+        time_sampling: scalar
+            number of samples per unit of time. or units of time per sample.
+            (digicam samples with 250MHz so a sample is 4ns long.)
+
+        sigma_e: scalar or 1-d array of shape (n_pixels, )
+            electronics noise (std-dev of normal distribution)
+            electronics noise is simulated as white noise.
+
+        jitter: scalar or 1d or 2d array of shape: ???
+            I do not know what kind of jitter is meant here, and in what
+            unit it might be given.
+
+        n_photon: sclar or 1d or 2d of shape: ???
```

Comment:
scalar would create same number of photons for all pixels. (signal photons affected by jitter)
1d with shape =(n_pixels,) would put each photon in each pixel
2d with shape = (n_pixels, ?) would be a list of photons per pixel